src/scala/microsoft-spark-2.3.x/src/main/scala/org/apache/spark/deploy/dotnet/DotnetRunner.scala

```diff
@@ -178,6 +178,10 @@ object DotnetRunner extends Logging {
   // permission to executable (only for Unix systems, since the zip file may have been
   // created under Windows. Finally, the absolute path for the executable is returned.
   private def resolveDotnetExecutable(dir: File, dotnetExecutable: String): String = {
+    if (Files.isRegularFile(Paths.get(dir.getAbsolutePath, dotnetExecutable))) {
+      return new File(dir, dotnetExecutable).getAbsolutePath
```

Comment:
If we return here, then we are not setting the permission below.

Do you think we can get rid of `walk.iterator` below and just test this directly here?

For example,
```
val resolved = Paths.get(dir.getAbsolutePath, dotnetExecutable)
if (!Files.isRegularFile(resolved)) {
   // throw here
}

// set permission

resolved
```

Reply:
I was thinking about this, but it will miss the scenario of `mySparkApp.exe` is under subfolder, where user doesn't specify the path. Like below:

```
spark-submit --class org.apache.spark.deploy.dotnet.DotnetRunner mySparkApp.zip mySparkApp.exe
```
which `mySparkApp.exe` is under subfolder after unzipping mySparkApp.zip

Reply:
Ah OK. In this case, let's just make sure setting permission code gets executed. So, check first if the file exists with the given path, if not fall back to `walk`, then set the permission.

Reply:
Sounds good. Changed!